copier/vcs.py

```diff
@@ -108,23 +111,61 @@ def checkout_latest_tag(local_repo: StrOrPath, use_prereleases: OptBool = False)
         return latest_tag
 
 
+def _perform_clone(src: str, dst: str, ref: OptStr = None) -> str:
+    """Clone from src into dst.
+
+    Args:
+        src:
+            Git-parseable repo clone source.
+        dst:
+            Git-parseable repo clone destination.
+        ref:
+            Reference to checkout. For Git repos, defaults to `HEAD`
+    """
+    _clone = git["clone", "--no-checkout", src, dst]
+    # Faster clones if possible
+    if GIT_VERSION >= Version("2.27"):
+        _clone = _clone["--filter=blob:none"]
+    _clone()
+
+    with local.cwd(dst):
+        git("checkout", ref or "HEAD")
+        git("submodule", "update", "--checkout", "--init", "--recursive", "--force")
+
+    return dst
```

Comment:
think this is needed so that `clone` returns `location` as is used [here](https://github.com/copier-org/copier/blob/3fe1f42728a4d3ce40b564d63420764f43d98ef1/copier/template.py#L434). I could also change `clone` to call `_perform_clone` and separately return `location`—lmk what you prefer